packages/subgraph-service/contracts/SubgraphService.sol

```diff
@@ -480,11 +481,14 @@ contract SubgraphService is
      *
      * @param _data Encoded data containing a signed RAV
      */
-    function _collectQueryFees(bytes memory _data) private returns (uint256 feesCollected) {
+    function _collectQueryFees(address indexer, bytes memory _data) private returns (uint256 feesCollected) {
         ITAPCollector.SignedRAV memory signedRav = abi.decode(_data, (ITAPCollector.SignedRAV));
-        address indexer = signedRav.rav.serviceProvider;
+        address serviceProvider = signedRav.rav.serviceProvider;
+        require(indexer == serviceProvider, SubgraphServiceInvalidIndexer(indexer));
         address allocationId = abi.decode(signedRav.rav.metadata, (address));
-        bytes32 subgraphDeploymentId = allocations.get(allocationId).subgraphDeploymentId;
+        Allocation.State memory allocation = allocations.get(allocationId);
+        require(allocation.indexer == indexer, SubgraphServiceInvalidAllocationIndexer(indexer, allocationId));
```

Comment:
I mean tokens collected get directly sent to the indexer specified in the RAV so they cant be stolen. And it would make no sense for an indexer to collect someone elses RAV because they would be locking their stake and get zero tokens out of it.

Reply:
As discussed I'll leave this check in case an indexer receives a voucher with a wrong allocationId so they don't accidentally lock in. their stake to collect another indexer's voucher. Also added a comment with this information.